ambry-account/src/main/java/com/github/ambry/account/AccountUpdater.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.commons.HelixNotifier;
+import com.github.ambry.commons.Notifier;
+import com.github.ambry.config.HelixPropertyStoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.Utils;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Properties;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import static com.github.ambry.account.AccountUtils.*;
+
+
+/**
+ * A component that is used to create/update {@link Account}s to {@code ZooKeeper} server. Underneath it
+ * employs an {@link AccountService} to do the actual write operation, and a {@link Notifier} to publish
+ * message for account metadata change once the write operation is completed.
+ */
+class AccountUpdater {
+  private static final int ZK_CLIENT_CONNECTION_TIMEOUT_MS = 5000;
+  private static final int ZK_CLIENT_SESSION_TIMEOUT_MS = 20000;
+  private final AccountService accountService;
+  private final Properties helixConfigProps = new Properties();
+  private final VerifiableProperties vHelixConfigProps;
+  private final HelixPropertyStoreConfig storeConfig;
+  private final Notifier notifier;
+
+  /**
+   * Constructor.
+   * @param zkServer The {@code ZooKeeper} server address to connect.
+   * @param storePath The path for {@link org.apache.helix.store.HelixPropertyStore}, which will be used as the
+   *                  root path for both {@link HelixAccountService} and {@link HelixNotifier}.
+   * @param zkConnectionTimeoutMs The timeout in millisecond to connect to the {@code ZooKeeper} server.
+   * @param zkSessionTimeoutMs The timeout in millisecond for a session to the {@code ZooKeeper} server.
+   */
+  private AccountUpdater(String zkServer, String storePath, int zkConnectionTimeoutMs, int zkSessionTimeoutMs) {
```

Comment:
I think that instead of having a class these could just be static utility methods used by `createOrUpdate`. The constructor can be a `getAccountService()` method that returns the `AccountService` directly and we can perform updates by directly calling `accountService.updateAccounts()`